### PR TITLE
refactor: Remove MPL library

### PR DESCRIPTION
**Pull request by stephenswat**
This commit removes the mostly unused MPL library. The few cases in which it was being used could be easily refactored into uses of fold expressions, which should allow better error messages.

### CODE REVIEW

**Review by andiwand — DISMISSED** [review]
😍

**Review by andiwand — APPROVED** [review]

**Review by paulgessinger — APPROVED** [review]
Awesome. Some of this are remnants from when I excised `boost::mpl` and reimplemented some stuff in C++14.

Happy to see this go!

File: Core/include/Acts/Utilities/detail/Extendable.hpp [hunk 1]
```diff
@@ -26,11 +24,11 @@ namespace Acts::detail {
 template <typename... extensions_t>
 struct Extendable {
   // clang-format off
-    static_assert(detail::all_of_v<std::is_default_constructible<extensions_t>::value...>,
+    static_assert((std::is_default_constructible<extensions_t>::value && ...),
```
Comment (AJPfleger) [inline Core/include/Acts/Utilities/detail/Extendable.hpp:27]:
```suggestion
    static_assert((std::is_default_constructible_v<extensions_t> && ...),
```
and the instances below

File: Core/include/Acts/Propagator/ActorList.hpp [hunk 2]
```diff
@@ -9,35 +9,35 @@
...
 template <typename... actors_t>
-  requires(
-      detail::all_of_v<std::is_default_constructible<actors_t>::value...> &&
-      detail::all_of_v<std::is_copy_constructible<actors_t>::value...> &&
-      detail::all_of_v<std::is_move_constructible<actors_t>::value...> &&
-      !detail::has_duplicates_v<actors_t...>)
+  requires((std::is_default_constructible<actors_t>::value && ...) &&
+           (std::is_copy_constructible<actors_t>::value && ...) &&
+           (std::is_move_constructible<actors_t>::value && ...) &&
+           ((Types::count<actors_t, TypeList<actors_t...>>::value == 1) && ...))
```
Comment (AJPfleger) [inline Core/include/Acts/Propagator/ActorList.hpp:30]: same here

**Review by AJPfleger — COMMENTED** [review]
I think, I could make sense to fix the marked sonarcloud issues. The others, I am fine with ignoring in this PR.

**Comment by stephenswat** [comment]
I'll take care of the SonarCloud issues. :+1:

**Comment by stephenswat** [comment]
Done; thanks for the suggestion.

**Comment by asalzburger** [comment]
That looks good, @AJPfleger - can you check again if you're happy! I like cleanup PRs very very much.

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [cd7810d67239e9c433185ec2e1f49e79b4be28eb]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963334)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963333)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963332)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963331)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963330)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963329)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963328)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963327)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963326)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963325)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963324)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963323)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963322)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963321)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963320)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963319)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963318)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963317)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963316)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963315)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963314)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963313)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963312)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963311)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963310)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963309)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963308)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963307)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963306)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963305)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963304)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963303)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43963302)||

**Pull request merged**